packages/insomnia-inso/src/commands/lint-specification.js

```diff
@@ -15,13 +18,33 @@ export async function lintSpecification(
 
   const specFromDb = identifier ? loadApiSpec(db, identifier) : await promptApiSpec(db, !!ci);
 
-  if (!specFromDb) {
-    logger.fatal(`Specification not found.`);
+  let specContent: string = '';
+
+  try {
+    if (specFromDb?.contents) {
+      logger.trace('Linting specification from database contents');
+      specContent = specFromDb.contents;
+    } else if (identifier) {
+      // try load as a file
+      const fileName = path.isAbsolute(identifier)
+        ? identifier
+        : path.join(workingDir || '.', identifier);
+      logger.trace(`Reading specification from file \`${fileName}\``);
+      try {
+        specContent = (await fs.promises.readFile(fileName)).toString();
+      } catch (e) {
+        throw new InsoError('Specification not found.');
+      }
+    } else {
+      throw new InsoError('Invalid specification.');
+    }
```

Comment:
Controlling flow through exceptions is generally discouraged as it is inefficient and difficult to follow. Re-throwing an external (such as a file read exception) with `InsoError` is useful because it provides additional context, but throwing an exception in this `else` block is avoidable.

You can simply log and exit:

```suggestion
    } else {
      logger.fatal(`Specification not found.`);
      return false;
    }
```

Reply:
Gotcha! I know some prefer using a single error state handling block and didn't think it was too much control flow manipulation, but I can certainly change.

Reply:
Ya fair enough! In this case it's easy enough to see because the `throw` and `catch` are so close, but usually that isn't the case. My preference is to keep the catch block only for genuine errors, and exit if we know we can exit.